pkg/alertmanager/operator.go

```diff
@@ -473,7 +474,14 @@ func (c *Operator) sync(key string) error {
 		return errors.Wrap(err, "making the statefulset, to update, failed")
 	}
 	if _, err := ssetClient.Update(sset); err != nil {
-		return errors.Wrap(err, "updating statefulset failed")
+		if fErr, ok := err.(*field.Error); ok && fErr.Type == field.ErrorTypeForbidden {
+			err := ssetClient.Delete(sset.GetName(), &metav1.DeleteOptions{})
+			if err != nil {
+				return errors.Wrap(err, "failed to delete StatefulSet to avoid forbidden action")
+			}
+		} else {
+			return errors.Wrap(err, "updating statefulset failed")
+		}
```

Comment:
also just a small stylistic nit to have a little less nested if's:
```
err := ssetClient.Update(sset)
fErr, ok := err.(*field.Error)

if ok && fErr.Type == field.ErrorTypeForbidden {
	if err := ssetClient.Delete(sset.GetName(), &metav1.DeleteOptions{}); err != nil {
		return errors.Wrap(err, "failed to delete StatefulSet to avoid forbidden action")
	}
}

if err != nil {
	return errors.Wrap(err, "updating statefulset failed")
}
```